### PR TITLE
Update Gazebo with new masses and sensor suite

### LINKED ISSUE #161: Connect gazebo sensors back to bzm
~~Blocked by #160~~

### LINKED ISSUE #196: Fix masses and inertials
Current in the car urdf, I reverted the mass of the car parts to be very small. The reason being we weren't able to see the car moving in gazebo. However, the long term solution is to uncomment out the `<!-- mass / -->` tags and remove the small-mass ones. Then, we'd need to modify the `max_torque` in `sim_car_interface.cpp` and potentially the `<inertials />` in urdf.

### PR DESCRIPTION
**Pull request by chsahit**
Closes #161
Closes #196

### CODE REVIEW

File: buzzmobile/simulation/models/car/car_joints.urdf [hunk 1]
```diff
@@ -198,7 +213,7 @@
             <headingDriftFrequency>0.0 0.0 0.0</headingDriftFrequency>
             <headingGaussianNoise>0.0 0.0 0.0</headingGaussianNoise>
         </plugin>
-    </gazebo>
+    </gazebo>-->
```
Comment (irapha) [inline buzzmobile/simulation/models/car/car_joints.urdf]: can you remove the commented out things?

**Review by irapha — APPROVED** [review]
im approving but there's a to-fix please fix <3

File: buzzmobile/simulation/models/car/car_joints.urdf (hunk 1 above)
Comment (chsahit) [reply buzzmobile/simulation/models/car/car_joints.urdf]: oops just saw this will do

Comment (irapha) [reply buzzmobile/simulation/models/car/car_joints.urdf]: cool. merge when you do, unless you add things then i can rereview

**Review by irapha — CHANGES_REQUESTED** [review]
unapproving since you asked me to. LMK when i can review

**Comment by chsahit** [comment]
@iRapha ok updated this PR because I was doing everything on this branch

File: buzzmobile/constants.yaml [hunk 2]
```diff
@@ -44,7 +44,7 @@ wheel_separation: 1.3716 # distance between left and right wheels in meters
 max_steering_angle: 0.262 # radians (=15 degrees)
 
 # Max speed of buzzmobile in m/s
-max_speed: 0.2
+max_speed: 2 #TODO this is a murder machine
```
Comment (irapha) [inline buzzmobile/constants.yaml]: why this change then? just for checking that simulation works?

File: buzzmobile/launch/includes/sim_spawn_car.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #! /bin/bash
 
-sleep 3
+sleep 15
```
Comment (irapha) [inline buzzmobile/launch/includes/sim_spawn_car.sh]: is this neeeddeeeeddddddddd??? such overhead, and we're not even using the launch script! why

File: buzzmobile/launch/simulation.launch [hunk 4]
```diff
@@ -9,7 +9,7 @@
         <include file="$(find buzzmobile)/launch/includes/mapping.launch"/>
         <include file="$(find buzzmobile)/launch/includes/steering.launch"/>
         <include file="$(find buzzmobile)/launch/includes/lidar.launch"/>
-        <include file="$(find buzzmobile)/launch/includes/controller.launch"/>
+        <!--<include file="$(find buzzmobile)/launch/includes/controller.launch"/>-->
```
Comment (irapha) [inline buzzmobile/launch/simulation.launch]: huh i guess this change makes sense, right? we don't want controller in simulation

wait controller.launch also has car_pose_mux, which we need to make this work. how did you test this at all without car_pose_mux?

File: buzzmobile/plan/steering/steering.py [hunk 5]
```diff
@@ -41,7 +41,7 @@
 BRAKING_DISTANCE = rospy.get_param('braking_distance')
 THRESHHOLD = rospy.get_param('braking_score_threshhold')
 BUZZMOBILE_WIDTH = rospy.get_param('buzzmobile_width')
-MAX_SPEED = rospy.get_param('max_speed', 1.0)
+MAX_SPEED = rospy.get_param('max_speed', 2.0) #TODO don't be a murder machine
```
Comment (irapha) [inline buzzmobile/plan/steering/steering.py]: maybe fix todo maybe?

File: buzzmobile/process/gps_mapper/gps_mapper.py [hunk 6]
```diff
@@ -134,6 +134,8 @@ def median_filter(fix):
     elif len(g['fixes']) % 2 and len(g['fixes']) > 1:
         return sorted_points[index]
     elif len(g['fixes']) > 1:
+        rospy.loginfo("sorted point: ")
+        rospy.loginfo(sorted_points)
```
Comment (irapha) [inline buzzmobile/process/gps_mapper/gps_mapper.py]: this shouldnt be in PR when its merged

File: buzzmobile/simulation/models/car/car_joints.urdf [hunk 7]
```diff
@@ -5,8 +5,8 @@
     <link name="chassis">
         <inertial>
             <origin xyz="0 0 0"/>
-            <mass value="5.0"/>
-            <inertia ixx="0.10595165" ixy="-0.002487144" ixz="0.011981253" iyy="0.126271962" iyz="0.011832006" izz="0.033185337"/>
+            <mass value="136.0"/>
+            <inertia ixx="39179.495" ixy="-257.203201" ixz="3771.586" iyy="95397.6332" iyz="-20.60398" izz="94045.9387"/>
```
Comment (irapha) [inline buzzmobile/simulation/models/car/car_joints.urdf:9]: YAY CORRECT INERTIALS?

File: buzzmobile/simulation/models/car/car_joints.urdf [hunk 8]
```diff
@@ -342,15 +356,21 @@
     <joint name="lidar_joint" type="fixed">
         <parent link="chassis"/>
         <child link="laser"/>
-        <origin xyz="0.255 0 0.110" rpy="0 0 0"/>
+        <origin xyz="0.255 0 5" rpy="0 0 0"/>
```
Comment (irapha) [inline buzzmobile/simulation/models/car/car_joints.urdf]: so all you did here was put the lidar way above the car. can you actually place it where our lidar is IRL?

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 9]
```diff
@@ -35,13 +35,13 @@ double speed_measured_right = 0.0;
 double steer_set_point = 0.0;
 
 // TODO(sahit): change to rosparams
-constexpr double chassis_length = 0.33246;
-constexpr double chassis_width = 0.28732;
+constexpr double chassis_length = 1.8; 
+constexpr double chassis_width = 1.57;
```
Comment (irapha) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: are these real-world?

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 10]
```diff
@@ -35,13 +35,13 @@ double speed_measured_right = 0.0;
...
 constexpr double inv_chassis_length = 1.0 / chassis_length;
 constexpr double chassis_width_2 = chassis_width / 2.0;
-constexpr double max_torque = 0.1;
+constexpr double max_torque = 2.5; //lololol
```
Comment (irapha) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: lol? no. rip

aka pls fix?

**Review by irapha — CHANGES_REQUESTED** [review]
also ur build is broken

File: buzzmobile/constants.yaml (hunk 2 above)
Comment (chsahit) [reply buzzmobile/constants.yaml]: our max speed is likely less than 2 m/s but also more than 0.2. I can update the TODO to be more professional (I'll do that for all of them in this PR), but max speed is something we just need to measure. If we left the max speed at 0.2 the car won't move.

File: buzzmobile/simulation/models/car/car_joints.urdf (hunk 7 above)
Comment (chsahit) [reply buzzmobile/simulation/models/car/car_joints.urdf:9]: yup! thanks to Matt and Wolfram mathworld!

File: buzzmobile/simulation/models/car/car_joints.urdf (hunk 8 above)
Comment (chsahit) [reply buzzmobile/simulation/models/car/car_joints.urdf]: yeah, its actually the same height as the base axle and about 12 inches from the front of the chassis.

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 9 above)
Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: yes but i think they have to be converted to imperial, good catch.

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 10 above)
Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: max torque is also just an estimate, but yeah I'll cleanup the comment.

**Comment by chsahit** [comment]
Not sure why builds are failing. According to the circle ci tests
`No SHA Files found, caching will not take place`
which doesnt mean a lot to me.
According to the simulation tests gazebo is timing out, don't know how to deal with that either.

**Comment by jgkamat** [comment]
The SHA files thing is just a missing config file, it dosen't cause a failure, it just means your builds are slower than they could be (in theory) #156 

I think this is the error you want to look at:

```
[0m[ INFO] [1493243463.256351398]: waitForService: Service [/gazebo/set_physics_properties] is now available.[0m
[0m[ INFO] [1493243463.878368355]: Laser Plugin (robotNamespace = /), Info: Using the 'robotNamespace' param: '/'[0m
[0m[ INFO] [1493243463.878766596]: Starting Laser Plugin (ns = /)![0m
spawn_model script started
[INFO] [WallTime: 1493243455.707552] Loading model xml from file
[INFO] [WallTime: 1493243455.708049] Waiting for service /gazebo/spawn_urdf_model
[INFO] [WallTime: 1493243463.255404] Calling service /gazebo/spawn_urdf_model
[INFO] [WallTime: 1493243463.902857] Spawn status: SpawnModel: Successfully spawned model
[1m[spawn_model-10] process has finished cleanly
log file: /home/developer/.ros/log/686fa93c-2aca-11e7-98b7-0242ac120002/spawn_model-10*.log[0m
[spawner-9] killing on exit
[gazebo-8] killing on exit
[buzzmobile/sim_car_interface-7] killing on exit
[buzzmobile/lidar_to_frame-6] killing on exit
[buzzmobile/steering-5] killing on exit
[buzzmobile/frame_merger-4] killing on exit
[buzzmobile/gps_mapper-2] killing on exit
[buzzmobile/bearing-1] killing on exit
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/home/developer/catkin_ws/src/buzzmobile/buzzmobile/sense/maps_querier/maps_querier.py", line 14, in <module>
    import googlemapskey as gmpskey
ImportError: No module named googlemapskey
[31m[buzzmobile/maps_querier-3] process has died [pid 35068, exit code 1, cmd /home/developer/catkin_ws/src/buzzmobile/buzzmobile/sense/maps_querier/maps_querier.py __name:=maps_querier __log:=/home/developer/.ros/log/686fa93c-2aca-11e7-98b7-0242ac120002/buzzmobile-maps_querier-3.log].
log file: /home/developer/.ros/log/686fa93c-2aca-11e7-98b7-0242ac120002/buzzmobile-maps_querier-3*.log[0m
libdc1394 error: Failed to initialize libdc1394
libdc1394 error: Failed to initialize libdc1394
libdc1394 error: Failed to initialize libdc1394
libdc1394 error: Failed to initialize libdc1394
Error opening log file: "/home/developer/.gazebo/gzserver.log"
Gazebo multi-robot simulator, version 2.2.3
Copyright (C) 2012-2014 Open Source Robotics Foundation.
Released under the Apache 2 License.
http://gazebosim.org

[1;32mMsg[0m Waiting for master[1;32mMsg[0m Connected to gazebo master @ http://127.0.0.1:11345
[1;32mMsg[0m Publicized address: 172.18.0.2
[1;31mError [RenderEngine.cc:665][0m Can't open display: 
[1;33mWarning [RenderEngine.cc:88][0m Unable to create X window. Rendering will be disabled
[1;33mWarning [ModelDatabase.cc:334][0m Getting models from[http://gazebosim.org/models/]. This may take a few seconds.
[1;31mError [CameraSensor.cc:85][0m Unable to create CameraSensor. Rendering is disabled.
[WARN] [WallTime: 1493243474.454752] Controller Spawner couldn't find the expected controller_manager ROS interface.
[31m[gazebo-8] escalating to SIGTERM[0m
========================== 1 failed in 42.20 seconds ===========================
```
From the simulation test.

Seems like a missing dependency, so not too big of a deal

File: buzzmobile/constants.yaml (hunk 2 above)
Comment (irapha) [reply buzzmobile/constants.yaml]: IIRC we put 0.2 down because the arduino software we got from RJ had a few bugs where in it would flush the speed update messages all at once, causing toyota mode to happen. We fixed it for when it was going forward, but didn't have time to do anything when it was turning.

You can experiment with this speed when you have access to the car, but I'd start with slow speed at first so you don't die while trying to figure out what the ideal speed is.

File: buzzmobile/simulation/models/car/car_joints.urdf (hunk 8 above)
Comment (irapha) [reply buzzmobile/simulation/models/car/car_joints.urdf]: Wait so this is.. correct? Or are you gonna update it?

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 9 above)
Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: uh what why omg are you sure?

**Comment by irapha** [comment]
As a note, @chsahit when you're looking for what caused a circleci build failure, you can either find the failure through the gh badges (see image)

screen shot 2017-05-09 at 10 28 41

or through the `runtests.sh` in circleci. This second one is a little misleading because the "error" technically occurs in the `exit.sh`, but trust

screen shot 2017-05-09 at 10 30 17

**Comment by jgkamat** [comment]
(also, you can see the pop up thingy for random commits through clicking on the check/x, or for random branches through the branches menu like this):
2017-05-09-103228_747x389_scrot

File: buzzmobile/launch/includes/sim_spawn_car.sh (hunk 3 above)
Comment (joshuamorton) [reply buzzmobile/launch/includes/sim_spawn_car.sh]: Its a good kludge, there should at least be a todo to investigate what that actually relies on, and to instrument something that manages process deps better.

File: buzzmobile/process/gps_mapper/gps_mapper.py (hunk 6 above)
Comment (joshuamorton) [reply buzzmobile/process/gps_mapper/gps_mapper.py]: That said, logging might be good.

Comment (irapha) [reply buzzmobile/process/gps_mapper/gps_mapper.py]: go away qa

File: buzzmobile/launch/includes/sim_spawn_car.sh (hunk 3 above)
Comment (irapha) [reply buzzmobile/launch/includes/sim_spawn_car.sh]: there's an issue #158

File: buzzmobile/process/gps_mapper/gps_mapper.py (hunk 6 above)
Comment (joshuamorton) [reply buzzmobile/process/gps_mapper/gps_mapper.py]: :3

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 9 above)
Comment (joshuamorton) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: Those look reasonable. 1.8 meters long and 1.57 meters wide.

Comment (joshuamorton) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: Wait hold on this doesn't at all match https://github.com/gtagency/buzzmobile/blob/master/buzzmobile/constants.yaml#L39 (which also says our wheels are 2.2 meters in circumference, which is wrong)

File: buzzmobile/simulation/models/car/car_joints.urdf (hunk 8 above)
Comment (chsahit) [reply buzzmobile/simulation/models/car/car_joints.urdf]: no its incorrect, that comment is more a note to myself.

Comment (irapha) [reply buzzmobile/simulation/models/car/car_joints.urdf]: Okay. Let me know when you have something I can review again.

File: buzzmobile/launch/simulation.launch (hunk 4 above)
Comment (chsahit) [reply buzzmobile/launch/simulation.launch]: you just have rqt publish the state of the car to be 0 (AUTO).

Comment (irapha) [reply buzzmobile/launch/simulation.launch]: but you still need car_pose_mux. That's the thing that takes in auto_car_pose and outputs car_pose.

I agree you don't need the controller *node*, but you need the car_pose_mux, which is in controller*.launch*

Comment (chsahit) [reply buzzmobile/launch/simulation.launch]: Yeah I'm not sure why this worked either because one would think that the PID controller would need to poll the pose of the car to calculate output, but it worked. Nonetheless, I added the car_pose_mux import back.

File: buzzmobile/constants.yaml (hunk 2 above)
Comment (chsahit) [reply buzzmobile/constants.yaml]: I'm bumping it down to 1 m/s now as a compromise. Still a bit faster than the car but if we set it to 0.2 then the simulation (running at 0.03 real time factor) doesn't let you see the car move.

File: buzzmobile/launch/includes/sim_spawn_car.sh (hunk 3 above)
Comment (chsahit) [reply buzzmobile/launch/includes/sim_spawn_car.sh]: I'll change it back to 3 since we're not using the launch script.

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 9 above)
Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: @iRapha yes gazebo only works with imperial and we want to be consistent.

Comment (joshuamorton) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: holy cow that's awful. Please be sure to track units in the comments (or var names, ie. `chassis_length_ft` just so that we don't have the problem of coming back in 6 months wondering what the hell 17.2 is.

Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: @chsahit please tell me you're joking. or its sarcarm i didn't pick up.

@joshuamorton +1000

Comment (chsahit) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: like 99% sure. but @joshuamorton yeah true good point

Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: nope its all metric, had it flipped in my head. it was autoCAD thats all imperial

Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: http://gazebosim.org/tutorials/?tut=ros_urdf#NoteOnUnits

Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: pls no hart atak anymor

Comment (chsahit) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: @joshuamorton yeah the dims don't match at all and these were the numbers I got from RJ...
In the CAD those are the dimensions of the car, when (if) I get access to the car, I'll check myself which measurements are correct. For now we should merge it in but with a TODO.

**Comment by irapha** [comment]
(and a gh issue)

File: buzzmobile/launch/simulation.launch [hunk 11]
```diff
@@ -9,6 +9,7 @@
         <include file="$(find buzzmobile)/launch/includes/mapping.launch"/>
         <include file="$(find buzzmobile)/launch/includes/steering.launch"/>
         <include file="$(find buzzmobile)/launch/includes/lidar.launch"/>
+        <node pkg="buzzmobile" name="car_pose_mux" type="car_pose_mux.py"/>
```
Comment (irapha) [inline buzzmobile/launch/simulation.launch]: I don't know how much I like this solution.. I'd rather have a redundant controller node started (and not doing anything) than launch nodes directly. Others can pitch in but in either case this change doesn't belong in this PR

File: buzzmobile/constants.yaml [hunk 12]
```diff
@@ -44,7 +44,7 @@ wheel_separation: 1.3716 # distance between left and right wheels in meters
 max_steering_angle: 0.262 # radians (=15 degrees)
 
 # Max speed of buzzmobile in m/s
-max_speed: 2 #TODO this is a murder machine
+max_speed: 1 #TODO: measure the max speed.
```
Comment (irapha) [inline buzzmobile/constants.yaml]: if TODO -> make gh issue
pls

Comment (irapha) [reply buzzmobile/constants.yaml]: I can sort the issues, i just want us to track TODOs as issues. It's fine for the "TODO" to exist in code too.

File: buzzmobile/plan/steering/steering.py [hunk 13]
```diff
@@ -41,7 +41,7 @@
 BRAKING_DISTANCE = rospy.get_param('braking_distance')
 THRESHHOLD = rospy.get_param('braking_score_threshhold')
 BUZZMOBILE_WIDTH = rospy.get_param('buzzmobile_width')
-MAX_SPEED = rospy.get_param('max_speed', 2.0) #TODO don't be a murder machine
+MAX_SPEED = rospy.get_param('max_speed', 1.0) #TODO measure car max speed
```
Comment (irapha) [inline buzzmobile/plan/steering/steering.py]: why does this `get_param` call have a second argument? Can you just remove it and have it rely on the real param? This way we don't have to modify `max_speed` in multiple places when we want to modify it

File: buzzmobile/simulation/models/car/car_joints.urdf [hunk 14]
```diff
@@ -356,7 +356,7 @@
     <joint name="lidar_joint" type="fixed">
         <parent link="chassis"/>
         <child link="laser"/>
-        <origin xyz="0.255 0 5" rpy="0 0 0"/>
+        <origin xyz="2 0 0" rpy="0 0 0"/>
```
Comment (irapha) [inline buzzmobile/simulation/models/car/car_joints.urdf]: is this the correct location of the lidar?

if not can you make a TODO and an issue to mesure it in the car exactly?

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 15]
```diff
@@ -39,7 +39,8 @@ constexpr double chassis_length = 1.8;
 constexpr double chassis_width = 1.57;
 constexpr double inv_chassis_length = 1.0 / chassis_length;
 constexpr double chassis_width_2 = chassis_width / 2.0;
-constexpr double max_torque = 2.5; //lololol
+//TODO(sahit): this can also be calculated/measured when we have the car
+constexpr double max_torque = 2.5;
```
Comment (irapha) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: make issue too

Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: All the measurement ones can be the same issue. Just be sure to link to each line where these TODOs exist

File: buzzmobile/launch/simulation.launch [hunk 16]
```diff
@@ -9,7 +9,8 @@
         <include file="$(find buzzmobile)/launch/includes/mapping.launch"/>
         <include file="$(find buzzmobile)/launch/includes/steering.launch"/>
         <include file="$(find buzzmobile)/launch/includes/lidar.launch"/>
-        <include file="$(find buzzmobile)/launch/includes/controller.launch"/>
+        <node pkg="buzzmobile" name="car_pose_mux" type="car_pose_mux.py"/>
+        <!--<include file="$(find buzzmobile)/launch/includes/controller.launch"/>-->
```
Comment (irapha) [inline buzzmobile/launch/simulation.launch]: I made a comment but idk why its not showing up. Id rather have a redundant controller node started than be launching nodes in two different ways in this file.

Others can argue otherwise, but in either case, this change doesnt belong in this PR (unless this is required to make sim work). If you feel strongly abt it then make a new PR pls

**Review by irapha — CHANGES_REQUESTED** [review]

File: buzzmobile/launch/simulation.launch (hunk 16 above)
Comment (irapha) [reply buzzmobile/launch/simulation.launch]: jk it was showing up

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 17]
```diff
@@ -35,13 +35,14 @@ double speed_measured_right = 0.0;
...
+constexpr double chassis_length = 1.8; 
+constexpr double chassis_width = 1.57;
 constexpr double inv_chassis_length = 1.0 / chassis_length;
 constexpr double chassis_width_2 = chassis_width / 2.0;
-constexpr double max_torque = 0.1;
+//TODO(sahit): this can also be calculated/measured when we have the car
+constexpr double max_torque = 2.5; 
 
-constexpr double wheel_circumference = 2.0 * M_PI * 0.036;
+constexpr double wheel_circumference = 2.0 * M_PI * 0.3302;
```
Comment (irapha) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:45]: dont forget that all of these should eventually be rosparams (and not constants in this file)

File: buzzmobile/plan/steering/steering.py (hunk 13 above)
Comment (chsahit) [reply buzzmobile/plan/steering/steering.py]: Yeah, i think the second parameter was there to prevent an exception being thrown if the rosparam server wasn't up or that rosparam wasn't there. But there exists a `get_param` call that only takes in the first parameter.

Comment (irapha) [reply buzzmobile/plan/steering/steering.py]: Yes. And I think that if rosparam server isn't up, this *should* throw an exception and error out.

**Comment by joshuamorton** [comment]
I was going to add that that seems reasonable, yeah.

File: buzzmobile/plan/steering/steering.py [hunk 18]
```diff
@@ -33,15 +33,15 @@
 PIXELS_PER_M = rospy.get_param('pixels_per_m')
 HEIGHT = rospy.get_param('image_height')
 WIDTH = rospy.get_param('image_width')
-MAX_ANGLE = rospy.get_param('max_steering_angle')
+MAX_ANGLE = rospy.get_param('max_steering_angle', 1.0)
```
Comment (irapha) [inline buzzmobile/plan/steering/steering.py:36]: i dont think this `, 1.0` is needed (see prev discussions)

**Review by irapha — APPROVED** [review]
Minor comment but I'm approving otherwise

**Comment by irapha** [comment]
bump

**Comment by joshuamorton** [comment]
Pls fix.

```
Traceback (most recent call last):
  File "/home/developer/catkin_ws/src/buzzmobile/buzzmobile/sense/maps_querier/maps_querier.py", line 14, in <module>
    import googlemapskey as gmpskey
ImportError: No module named googlemapskey
```

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 19]
```diff
@@ -106,8 +106,13 @@ int main(int argc, char **argv) {
     ros::Publisher leftSteeringPublisher = handle.advertise<std_msgs::Float64>("/left_steer_position_controller/command", 1);
     ros::Publisher rightSteeringPublisher = handle.advertise<std_msgs::Float64>("/right_steer_position_controller/command", 1);
 
+<<<<<<< HEAD
```
Comment (joshuamorton) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: Please fix.

File: buzzmobile/tests/simulation/test_gazebo_connection.py [hunk 20]
```diff
@@ -1,14 +1,17 @@
...
     def test_clock_runs(self):
         """Test that simulation.launch correctly launches gazebo"""
-        with check_topic('/clock', Clock) as ct:
+        with self.check_topic('/clock', Clock) as ct:
             assert(ct.message)
-        with check_topic('/clock', Clock) as ct:
-            assert(ct.message)
-
+            print(ct.message)
+            time.sleep(30)
```
Comment (joshuamorton) [inline buzzmobile/tests/simulation/test_gazebo_connection.py]: Try moving this outside of the with block and see what happens/if you can shorten it.

**Review by joshuamorton — COMMENTED** [review]
Go ahead and pull from master to clean up this PR some.

**Comment by chsahit** [comment]
@joshuamorton  I pulled from master and git reports that my branch is already up to date

**Comment by joshuamorton** [comment]
Weird, its giving some really unhelpful diffs that include already merged changes.

**Comment by chsahit** [comment]
So I didn't change anything but I went to the circle CI console and hit build and now we're passing... @joshuamorton @iRapha

**Comment by joshuamorton** [comment]
It may be because I pushed to pyros.

**Comment by irapha** [comment]
holy guacamoli

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 9 above)
Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:39]: I made an issue for this #208

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp (hunk 17 above)
Comment (irapha) [reply buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp:45]: There's an issue out for this #190 so this PR can be merged

**Comment by irapha** [comment]
Feel free to merge. But before you do, please double check this PR's description to make sure it actually closes the issues listed
